src/storage.rs

```diff
@@ -2291,6 +2292,16 @@ impl Cache {
         guard.command_history.last_fetch = Some(last_fetch);
     }
 
+    pub fn get_last_fetch_mode(&self) -> Option<FetchMode> {
+        let guard = self.state.lock().unwrap();
+        guard.command_history.last_fetch_mode
+    }
+
+    pub fn set_last_fetch_mode(&self, mode: FetchMode) {
+        let mut guard = self.state.lock().unwrap();
+        guard.command_history.last_fetch_mode = Some(mode);
+    }
+
```

Comment:
I'd like to see (either in addition to, or replacing these in the `pub` interface):
```rust
pub fn fetch_mode(&self, chosen_mode: Option<FetchMode>) -> FetchMode {
    if let Some(mode) = chosen_mode {
        self.set_last_fetch_mode(mode);
    }
    self.get_last_fetch_mode().unwrap_or(FetchMode::DiffRs)
}
```
This deduplicates the logic, however my main reasoning is to have one place where the default `FetchMode` is established. And/or we could add a `impl Default for FetchMode`.

Reply:
I centralized & documented the logic a bit more, also moving the local-forcing stuff into the function a bit to make it more explicit.